ledger/participant-integration-api/src/main/scala/platform/store/backend/common/TemplatedStorageBackend.scala

```diff
@@ -870,10 +887,11 @@ private[backend] object TemplatedStorageBackend {
       witnessesWhereClause: String,
       submittersInPartiesClause: String,
       filteredWitnessesClause: String,
+      columnEqualityBoolean: String,
   )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]] = {
     import com.daml.platform.store.Conversions.ledgerStringToStatement
     SQL"""select #$selectColumnsForTransactionTree, #$filteredWitnessesClause as event_witnesses,
-                 event_kind = 20 as exercise_consuming,
+                 #$columnEqualityBoolean as exercise_consuming,
```

Comment:
Did you check whether the query plan for
```
SELECT *
FROM participant_events
WHERE event_kind = 20
```
is equal to the one for
```
SELECT *
FROM participant_events_create
```
?

Reply:
this query is just returning a boolean for whether the column value is 20 though, since oracle does not have a native boolean type for a column return value -- it's not actually checking the validity of the view -- the view created for oracle is basically identical to what is done for postgres/h2

Reply:
I understand that, I was more interested to see whether Oracle is smart enough to realize that it only needs to access the `participant_events_create` table and not the whole `participant_events` view (which is a union of all event partitions).

We could simplify a lot of queries using the above mentioned substitution, but so far there were not many arguments for doing it.

Reply:
@rautenrieth-da I get your point, it might be worth double-checking it, but this is a wrong query for that here the event_kind = 20 populates the excercise_consuming field which is emitted to the resultset.

Reply:
@rautenrieth-da when you query via the view oracle does a full table scan of all the tables associated with the view - so all the selects and union alls across all the tables.  when you select * from participant_events_create it does full table scan of just that table

Reply:
when i query with an indexed field it does hit the indices of all the underlying tables behind the view though
![image](https://user-images.githubusercontent.com/25179017/123493425-66482600-d5ea-11eb-9ca1-6bd434c2e95a.png)

Reply:
i'm going to mark this as resolved -- if we want to do further optimization with view vs direct table queries let's do as a follow-on and try to approach it in a cross-db way

Reply:
:+1: